### DISCUSSION TITLE
Hard delete users

### DISCUSSION THREAD

**Discussion by ErwinLiemburg** [directus/directus#2750] [Archive]
For an upcoming project I have to create where frontend website users are able to create an user account. Creating that account isn't the problem, removing the account is. The actual removal of the account isn't that hard, but accounts aren't really removed. They are soft deleted, due to accountability reasons.
On soft deleting an account, the account is disabled, but information remains in place, amongst which is the users e-mail address that is required for the account.

In most cases the will remain for a few months, but have to be removed after a few months. according to Dutch privacy law (correct me if I'm wrong) all information relevant to a person has te be removed after a certain time when it is no longer relavant, or when the person in question askes for it (or manually closes the account). Due to the fact that the account can only be soft deleted, the e-mail address amongst others remains in the system which conflicts with the Dutch privacy law.

Is there a solution to hard delete user accounts, or does anybody have another solution where I can create user accounts that can be removed when needed?

**Comment by benhaynes** [comment]
Seems like you would want to use an [event hook](https://docs.directus.io/extensions/hooks.html) with [direct data access](https://docs.directus.io/advanced/api/data.html) to hard delete the users. This may be solved in v9 — I'll discuss with the team.

**Comment by rijkvanzanten** [comment]
It's the admin app that soft-deletes items, you can still hard delete through the API too

**Comment by ErwinLiemburg** [comment]
That sounds promising. I'll check it out. Thanks!
